read_files.cpp

```diff
@@ -280,6 +280,7 @@ QMap<QString, std::vector<Sensor::ButtonMap>> loadButtonMapsJson(const QJsonDocu
                     for (auto i = buttonMapArr.constBegin(); i != buttonMapArr.constEnd(); ++i)       // Loop through button map items
                     {
                         QJsonValue val = *i;
+                        mapItem++;
```

Comment:
`mapItem` is incremented again at the end of the else branch.

Shouldn't it be enough to just put this into the for loop definition:

`for (auto i = buttonMapArr.constBegin(); i != buttonMapArr.constEnd(); ++i, mapItem++)`

I'd also propose to make the type `int mapItem = 0;`

Reply:
Done